include/zephyr/bluetooth/l2cap.h

```diff
@@ -557,6 +558,52 @@ int bt_l2cap_ecred_chan_connect(struct bt_conn *conn,
  */
 int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu);
 
+/** @brief Reconfigure Enhanced Credit Based L2CAP channels
+ *
+ *  Experimental API to reconfigure L2CAP ECRED channels with explicit MPS and
+ *  MTU values.
+ *
+ *  Pend a L2CAP ECRED reconfiguration for up to 5 channels. All provided
+ *  channels must share the same @ref bt_conn.
+ *
+ *  Cannot decrease MTU of any channel. And, cannot decrease MPS of any channel
+ *  when more than one channel is provided.
+ *
+ *  There is no dedicated callback for this operation, but whenever a peer responds to a
+ *  reconfiguration request, each channel's reconfigured() callback is invoked.
+ *
+ *  This function may block for the duration of L2CAP_RTX_TIMEOUT.
```

Comment:
Yikes; this almost feel like it deserves a @warning or at least a @note. 

With the blocking changes we've seen in the system workqueue, we definitely do not want this to be called in the system workqueue then, if it can block for 2 seconds.

`L2CAP_RTX_TIMEOUT` is not a public available value. I think we should not refer to internal values like `L2CAP_RTX_TIMEOUT` in the public API. Should we move that to `l2cap.h` and `@ref` it?

Reply:
Maybe we should instead just say it can block for an unspecified amount of time? I think so now. Apparently `L2CAP_RTX_TIMEOUT` is implementation specific, and it looks like we chose `L2CAP_RTX_TIMEOUT` somewhat arbitrarily as the timeout for the allocation. It could change in the future.

Ideally this API should not block at all, and we should ensure this in future work. For now, this is mostly a copy-paste of `bt_l2cap_ecred_chan_reconfigure`.

Reply:
> Maybe we should instead just say it can block for an unspecified amount of time? 

I think that's OK for now.